src/main/java/de/tum/in/www1/artemis/service/connectors/athena/AthenaRepositoryExportService.java

```diff
@@ -0,0 +1,117 @@
+    private Path repoDownloadClonePath;
+
+    private final ProgrammingExerciseRepository programmingExerciseRepository;
+
+    private final ProgrammingExerciseExportService programmingExerciseExportService;
+
+    private final FileService fileService;
+
+    private final ProgrammingSubmissionRepository programmingSubmissionRepository;
+
+    public AthenaRepositoryExportService(ProgrammingExerciseRepository programmingExerciseRepository, ProgrammingExerciseExportService programmingExerciseExportService,
+            FileService fileService, ProgrammingSubmissionRepository programmingSubmissionRepository) {
+        this.programmingExerciseRepository = programmingExerciseRepository;
+        this.programmingExerciseExportService = programmingExerciseExportService;
+        this.fileService = fileService;
+        this.programmingSubmissionRepository = programmingSubmissionRepository;
+    }
+
+    /**
+     * Check if feedback suggestions are enabled for the given exercise, otherwise throw an exception.
+     *
+     * @param exercise the exercise to check
+     * @throws AccessForbiddenException if the feedback suggestions are not enabled for the given exercise
+     */
+    private void checkFeedbackSuggestionsEnabledElseThrow(Exercise exercise) {
+        if (!exercise.getFeedbackSuggestionsEnabled()) {
+            log.error("Feedback suggestions are not enabled for exercise {}", exercise.getId());
+            throw new AccessForbiddenException("Feedback suggestions are not enabled for exercise");
+        }
+    }
+
+    /**
+     * Export the repository for the given exercise and participation to a zip file.
+     * The ZIP file will be deleted automatically after 15 minutes.
+     *
+     * @param exerciseId     the id of the exercise to export the repository for
+     * @param submissionId   the id of the submission to export the repository for (only for student repository, otherwise pass null)
+     * @param repositoryType the type of repository to export. Pass null to export the student repository.
+     * @return the zip file containing the exported repository
+     * @throws IOException              if the export fails
+     * @throws AccessForbiddenException if the feedback suggestions are not enabled for the given exercise
+     */
+    public File exportRepository(long exerciseId, Long submissionId, RepositoryType repositoryType) throws IOException {
+        log.debug("Exporting repository for exercise {}, submission {}", exerciseId, submissionId);
+
+        var programmingExercise = programmingExerciseRepository.findByIdElseThrow(exerciseId);
+        checkFeedbackSuggestionsEnabledElseThrow(programmingExercise);
+
+        var exportOptions = new RepositoryExportOptionsDTO();
+        exportOptions.setAnonymizeRepository(false);
+        exportOptions.setExportAllParticipants(false);
+        exportOptions.setFilterLateSubmissions(false);
+        exportOptions.setFilterLateSubmissionsIndividualDueDate(false);
```

Comment:
Considering that this will always be called after the exercise due date, I think it's OK to only filter the late pushed commits here and assume there were no changes after that. This way, it doesn't matter that we use the participation ID.

However, I still missed an option and you are right about the repository anonymization. The new config is this:
```java
exportOptions.setAnonymizeRepository(true);
exportOptions.setExportAllParticipants(false);
exportOptions.setFilterLateSubmissions(true);
exportOptions.setFilterLateSubmissionsDate(programmingExercise.getDueDate());
exportOptions.setFilterLateSubmissionsIndividualDueDate(false); // Athena currently does not support individual due dates
```